src/ast/passes/resource_analyser.cpp

```diff
@@ -242,6 +258,14 @@ void ResourceAnalyser::visit(Call &call)
 
     resources_.skboutput_args_.emplace_back(file.str, offset.n);
     resources_.needs_perf_event_map = true;
+  } else if (call.func == "delete") {
```

Comment:
Is this to support the old/deprecated `delete` API where the map and key are passed as one arg e.g. `delete(@mymap[1])`? If so, I don't think we should add this logic here or else heavily document that this should get removed.

The new API has a single value as the second arg and should already be accounted for (in terms of global buffer vs stack).

Reply:
I will fix this to not loop through and only look at the first arg. I just copied codegenllvm module previously

Reply:
Fixed in latest

Reply:
Is the map.type already updated at this point in the case the second arg is larger then what's currently stored in the map? Example:
```
@a[1, "by"] = 1;
delete(@a, (1, "hellolongerstr"));
```